test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs

```diff
@@ -976,6 +977,197 @@ public static CommandRunnerResult RunRestore(SimpleTestPathContext pathContext,
+            <namespace id=""External.*"" />
+            <namespace id=""Others.*"" />
+        </packageSource>
+        <packageSource key=""ContosoRepository"">
+            <namespace id=""Contoso.*"" />             
+            <namespace id=""Test.*"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+
+                var ContosoReal = new SimpleTestPackageContext()
+                {
+                    Id = "Contoso.A",
+                    Version = "1.0.0"
+                };
+                ContosoReal.AddFile("lib/net461/contosoA.dll");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    contosoRepositoryPath,
+                    PackageSaveMode.Defaultv3,
+                    ContosoReal);
+
+                var ExternalA = new SimpleTestPackageContext()
+                {
+                    Id = "Contoso.A",  // Initial version had package id conflict with Contoso repository
+                    Version = "1.0.0"
+                };
+                ExternalA.AddFile("lib/net461/externalA.dll");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    externalRepositoryPath,
+                    PackageSaveMode.Defaultv3,
+                    ExternalA);
+
+                var ExternalB = new SimpleTestPackageContext()
+                {
+                    Id = "External.B",  // name conflict resolved.
+                    Version = "2.0.0"
+                };
+                ExternalB.AddFile("lib/net461/externalB.dll");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    externalRepositoryPath,
+                    PackageSaveMode.Defaultv3,
+                    ExternalB);
+
+                Util.CreateFile(Path.GetDirectoryName(projectA.ProjectPath), "packages.config",
+@"<packages>
+  <package id=""Contoso.A"" version=""1.0.0"" targetFramework=""net461"" />
+  <package id=""External.B"" version=""2.0.0"" targetFramework=""net461"" />
+</packages>");
+
+                solution.Projects.Add(projectA);
+                solution.Create(pathContext.SolutionRoot);
+
+                // Act
+                var result = RunRestore(pathContext, _successExitCode);
+
+                // Assert
+                var contosoRestorePath = Path.Combine(projectAPackages, ContosoReal.ToString(), ContosoReal.ToString() + ".nupkg");
+                var externalRestorePath = Path.Combine(projectAPackages, ExternalB.ToString(), ExternalB.ToString() + ".nupkg");
+                using (var nupkgReader = new PackageArchiveReader(contosoRestorePath))
+                {
+                    var allFiles = nupkgReader.GetFiles().ToList();
+                    // Assert correct Contoso package from Contoso repository was restored.
+                    Assert.Contains("lib/net461/contosoA.dll", allFiles);
+                }
+                Assert.True(File.Exists(externalRestorePath));
```

Comment:
Sample output:

```
NuGet Version: 6.0.0.20211
Restoring NuGet packages for solution C:\NuGetProj\NuGet.Client\.test\work\f7100b08\70daf15d\solution\solution.sln.
MSBuild auto-detection: using msbuild version '17.0.0.30702' from 'C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\bin'. Use option -MSBuildVersion to force nuget to use a specific version of MSBuild.
MSBuild P2P timeout [ms]: 120000
C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\bin\msbuild.exe "C:\Users\eryondon\AppData\Local\Temp\NuGetScratch\3h52niri.mlg.nugetinputs.targets" /t:GenerateRestoreGraphFile /nologo /nr:false /v:q /p:NuGetRestoreTargets="C:\Users\eryondon\AppData\Local\Temp\NuGetScratch\xobx0vp2.uo2.nugetrestore.targets" /p:RestoreUseCustomAfterTargets="True" /p:RestoreTaskAssemblyFile="C:\NuGetProj\NuGet.Client\test\NuGet.Clients.FuncTests\NuGet.CommandLine.FuncTest\bin\Debug\net472\NuGet\NuGet.exe" /p:RestoreSolutionDirectory="C:\NuGetProj\NuGet.Client\.test\work\f7100b08\70daf15d\solution\\" /p:SolutionDir="C:\NuGetProj\NuGet.Client\.test\work\f7100b08\70daf15d\solution\\" /p:SolutionName="solution"
Restoring NuGet package Contoso.A.1.0.0.
Package namespace prefix matches found for package id 'Contoso.A' are: 'contosorepository'
Package namespace: Skipping source 'C:\Users\eryondon\.nuget\packages\' for package id 'Contoso.A'
Package namespace: Skipping source 'ExternalRepository' for package id 'Contoso.A'
Package namespace: Trying source 'ContosoRepository' for package id 'Contoso.A'
Adding package 'Contoso.A.1.0.0' to folder 'C:\NuGetProj\NuGet.Client\.test\work\f7100b08\70daf15d\solution\packages'
Added package 'Contoso.A.1.0.0' to folder 'C:\NuGetProj\NuGet.Client\.test\work\f7100b08\70daf15d\solution\packages'
Added package 'Contoso.A.1.0.0' to folder 'C:\NuGetProj\NuGet.Client\.test\work\f7100b08\70daf15d\solution\packages' from source 'C:\NuGetProj\NuGet.Client\.test\work\f7100b08\70daf15d\solution\ContosoRepository'
Restoring NuGet package External.B.2.0.0.
Package namespace prefix matches found for package id 'External.B' are: 'externalrepository'
Package namespace: Skipping source 'C:\Users\eryondon\.nuget\packages\' for package id 'External.B'
Package namespace: Trying source 'ExternalRepository' for package id 'External.B'
Package namespace: Skipping source 'ContosoRepository' for package id 'External.B'
Adding package 'External.B.2.0.0' to folder 'C:\NuGetProj\NuGet.Client\.test\work\f7100b08\70daf15d\solution\packages'
Added package 'External.B.2.0.0' to folder 'C:\NuGetProj\NuGet.Client\.test\work\f7100b08\70daf15d\solution\packages'
Added package 'External.B.2.0.0' to folder 'C:\NuGetProj\NuGet.Client\.test\work\f7100b08\70daf15d\solution\packages' from source 'C:\NuGetProj\NuGet.Client\.test\work\f7100b08\70daf15d\solution\ExternalRepository'
NuGet Config files used:
    C:\NuGetProj\NuGet.Client\.test\work\f7100b08\70daf15d\NuGet.Config
    C:\NuGetProj\NuGet.Client\NuGet.Config
    C:\Users\eryondon\AppData\Roaming\NuGet\NuGet.Config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.FallbackLocation.config
    C:\Program Files (x86)\NuGet\Config\Microsoft.VisualStudio.Offline.config
Feeds used:
    C:\Users\eryondon\.nuget\packages\
    C:\NuGetProj\NuGet.Client\.test\work\f7100b08\70daf15d\solution\ExternalRepository
    C:\NuGetProj\NuGet.Client\.test\work\f7100b08\70daf15d\solution\ContosoRepository
Installed:
    2 package(s) to packages.config projects

```

Reply:
As far as the package namespace messages go, I think the only message necessary is:

> Package namespace prefix matches found for package id 'Contoso.A' are: 'contosorepository'

Reply:
Sure. I removed `Package namespace: Skipping source ..` and `Package namespace: Trying source...`.
If in the future someone raise concern about not enough logging then adding back is not hard.
How about `Package namespace prefix match not found for package id 'Contoso.A'`? 
Currently we have following if package doesn't match any prefix.

https://github.com/NuGet/NuGet.Client/blob/d4f9aae7f32435844ae56878100cb4c5dae06129/src/NuGet.Core/NuGet.PackageManagement/PackageDownloader.cs#L111-L114